src/Uno.UI/UI/Xaml/Controls/Popup/Popup.wasm.cs

```diff
@@ -61,23 +61,36 @@ protected override void OnIsOpenChanged(bool oldIsOpen, bool newIsOpen)
 				}
 			}
 
+			bool pressed = false;
+
 			if (previousPanel != null)
 			{
-				previousPanel.PointerPressed -= Panel_PointerPressed;
+				previousPanel.PointerPressed -= pointerPressed;
+				previousPanel.PointerReleased -= pointerReleased;
 			}
 			if (newPanel != null)
 			{
-				newPanel.PointerPressed += Panel_PointerPressed;
+				newPanel.PointerPressed += pointerPressed;
+				newPanel.PointerReleased += pointerReleased;
 			}
-		}
 
-		private void Panel_PointerPressed(object sender, Input.PointerRoutedEventArgs e)
-		{
-			if (IsLightDismissEnabled)
+			void pointerPressed(object sender, Input.PointerRoutedEventArgs args)
 			{
-				IsOpen = false;
+				// Both pressed & released must reach
+				// the popup to close it.
+				// (and, obviously, the popup must be light dismiss!)
+				pressed = IsLightDismissEnabled;
```

Comment:
You can't just use Tapped intead of PointerReleased?

Reply:
Tapped is not a "true" routed event and is the result of the interpretation of pointer events.
I want to prevent source events from generating the tap.